pkgs/development/compilers/llvm/17/mlir/gnu-install-dirs.patch

```diff
@@ -0,0 +1,15 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index c91e9cd93dc8..23b6032a46b7 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -32,8 +32,8 @@ if(MLIR_STANDALONE_BUILD)
+   endif()
+ 
+   set(CMAKE_LIBRARY_OUTPUT_DIRECTORY
+-    "${CMAKE_CURRENT_BINARY_DIR}/lib${LLVM_LIBDIR_SUFFIX}")
+-  set(CMAKE_RUNTIME_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/bin")
```

Comment:
I don't know to be honest, I followed the same pattern from other llvm components, I'll try again without this and if it works I'll remove the change. Otherwise, I'll keep it and mark it as resolved

Reply:
OK sounds good.

Yes some of the other patches are a bit stale as opposed to what I actually upstreamed, and some things maybe were intentionally not upstreamed because it turned out they weren't needed.

Reply:
I removed this patch and built mlir. While it builds, but programs that target mlir fail to find it. To be exact, cmake fails to find `mlir-tblgen` but with this patch the same program compiles with no issue. So I think we need it.

Reply:
Can you write a test for this? OK if it is in a follow-up PR.

Reply:
for sure